docs/userguide/ui-concepts.md

```diff
@@ -1,3 +1,151 @@
+      Text("message")
+    }
+    ```
+
+After all the chest-thumping above about [Separation of Concerns](#separation-of-concerns), these code samples probably don't look very separate. That's because, while the libraries themselves are completely decoupled, they don't force that strict rigor on your app code.
+
+If you aren't building, say, a core workflow module that you want to ship separately from its Android app, and an alternative command line interface, you'd probably gain nothing from enforced separation but boilerplate and runtime errors.
+And in practice, your Workflow unit tests won't call `viewControllerDescription()` or `viewFactory` and will build and run just fine against the JVM.
+
+If you're one of the few who really and trully do need impermeable boundaries between your core and UI modules, that's not hard to get. Your `Screen` implementations can be defined completely separately from their view code and bound later.
+
+
+=== "iOS"
+    ```Swift
+    // WelcomeScreen.swift
+    struct WelcomeScreen {
+        var name: String
+        var onNameChanged: (String) -> Void
+        var onLoginTapped: () -> Void
+    }
+
+    // WelcomeViewController.swift
+    extension WelcomeScreen: Screen {
+        func viewControllerDescription(environment: ViewEnvironment) -> ViewControllerDescription {
+            return WelcomeViewController.description(for: self, environment: environment)
+        }
+    }
+
+    private final class WelcomeViewController: ScreenViewController<WelcomeScreen> {
+        // ...
+    ```
+
+=== "Android"
+    ```Kotlin
+    // HelloScreen.kt
+    data class HelloScreen(
+      val message: String,
+      val onClick: () -> Unit
+    ) : Screen
+
+    // HelloScreenViewFactory.kt
+    object HelloScreenViewFactory: ScreenViewFactory<HelloScreen>
+    by ScreenViewFactory.fromViewBinding(HelloViewBinding::inflate) { r, _ ->
+          helloMessage.text = r.message
+          helloMessage.setOnClickListener { r.onClick() }
+        }
+    }
+
+    // HelloActivity.kt
+    class HelloWorkflowActivity : AppCompatActivity() {
+      override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val model: HelloViewModel by viewModels()
+        setContentView(
+          WorkflowLayout(this).apply { take(lifecycle, model.renderings) }
+        )
+      }
+    }
+
+    class HelloViewModel(savedState: SavedStateHandle) : ViewModel() {
+      private val viewRegistry = ViewRegistry(HelloScreenViewFactory)
+
+      val renderings: StateFlow<Screen> by lazy {
+        renderWorkflowIn(
+          workflow = HelloWorkflow.mapRenderings { it.withRegistry(viewRegistry) },
+          scope = viewModelScope,
+          savedStateHandle = savedState
+        )
+      }
+    }
+    ```
+
+!!! faq "Why \"Screen\"?"
+    We chose the name "Screen" because "View" would invite confusion with the like-named Android and iOS classes, and because "Box" didn't occur to us. 
```

Comment:
Is it worth calling out that this is also unrelated to UIScreen, which is UIKit's representation of a hardware display?

Reply:
Missed this before I merged. Maybe add this?

> (No one seems to have been confused by the fact that `Screen` and iOS's `UIScreen` are unrelated.)

Reply:
Yeah, I think so. Thanks!